linera-base/src/tracing_otel.rs

```diff
@@ -59,15 +57,17 @@ pub fn init_with_opentelemetry(log_name: &str, otlp_endpoint: Option<&str>) {
     global::set_tracer_provider(tracer_provider.clone());
     let tracer = tracer_provider.tracer("linera");
 
-    let telemetry_only_filter =
-        filter_fn(|metadata| metadata.is_span() && metadata.target() == "telemetry_only");
+    let otel_filter = filter_fn(|metadata| {
+        if !metadata.is_span() {
+            return false;
+        }
+        metadata
+            .fields()
+            .field("otel.skip")
+            .is_none()
```

Comment:
Nit: it would be nice to require the value be true in order to disable; this makes it easier to do more complex conditionals, like
```rust
#[tracing::instrument(fields(otel.skip = !otel))]
fn foo(otel: bool) {
  …
}
```